lib/api/BackbeatAPI.js

```diff
@@ -1260,40 +1260,12 @@ class BackbeatAPI {
             this._healthcheck = new Healthcheck(this._repConfig, this._zkClient,
                 this._crrProducer, this._crrStatusProducer,
                 this._metricProducer);
-            this._locationStatusManager = new LocationStatusManager(
-                this._mongoClient,
-                this._zkClient,
-                this._redisPublisher,
-                {
-                    crr: {
-                        namespace: zookeeperReplicationNamespace,
-                        statePath: zkReplicationStatePath,
-                        topic: this._crrTopic,
-                        isMongo: false,
-                    },
-                    ingestion: {
-                        namespace: zookeeperIngestionNamespace,
-                        statePath: zkIngestionStatePath,
-                        topic: this._ingestionTopic,
-                        isMongo: false,
-                    },
-                    lifecycle: {
-                        isMongo: true,
-                    }
-                },
-                this._logger,
-            );
-            return this._locationStatusManager._setupLocationStatusStore(err => {
-                if (err) {
-                    this._logger.error('error setting up location status manager', {
-                        method: 'BackbeatAPI.setupInternals',
-                        error: err.message,
-                    });
-                    return cb(err);
-                }
-                this._logger.info('BackbeatAPI setup ready');
+            if (this._config.queuePopulator.mongo) {
+                return this._setupLocationStatusManager(cb);
+            } else {
```

Comment:
we still need to initialize the `LocationStatusManager` as the pause/resume of crr uses redis/zookeeper and not mongodb and so we need to keep it. This case should be handled inside the `LocationStatusManager` class.